docs/how-to/stream_ordered_allocator.rst

```diff
@@ -0,0 +1,423 @@
+
+.. code-block::cpp
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    int main() {
+        hipMemPool_t memPool;
+        hipDevice_t device = 0; // Specify the device index
+
+        // Create a memory pool.
+        hipMemPoolCreate(&memPool, 0, 0);
+
+        // Allocate memory from the pool (e.g., 1 MB).
+        size_t allocSize = 1 * 1024 * 1024;
+        void* ptr;
+        hipMalloc(&ptr, allocSize);
+
+        // Free the allocated memory.
+        hipFree(ptr);
+
+        // Trim the memory pool to a specific size (e.g., 512 KB).
+        size_t newSize = 512 * 1024;
+        hipMemPoolTrimTo(memPool, newSize);
+
+        // Clean up.
+        hipMemPoolDestroy(memPool);
+
+        std::cout << "Memory pool trimmed to " << newSize << " bytes." << std::endl;
+        return 0;
+    }
+
+
+Resource usage statistics
+-------------------------
+
+Resource usage statistics can help in optimization. The following pool attributes to query memory usage:
+
+- ``hipMemPoolAttrReservedMemCurrent`` returns the current total physical GPU memory consumed by the pool.
+- ``hipMemPoolAttrUsedMemCurrent`` returns the total size of all memory allocated from the pool.
+- ``hipMemPoolAttrReservedMemHigh`` returns the total physical GPU memory consumed by the pool since the last reset.
+- ``hipMemPoolAttrUsedMemHigh`` returns the all memory allocated from the pool since the last reset.
+
+You can reset them to the current value using the ``hipMemPoolSetAttribute()``.
+
+.. code-block::cpp
+
+    #include <hip/hip_runtime.h>
+
+    // sample helper functions for getting the usage statistics in bulk
+    struct usageStatistics {
+        uint64_t reservedMemCurrent;
+        uint64_t reservedMemHigh;
+        uint64_t usedMemCurrent;
+        uint64_t usedMemHigh;
+    };
+
+    void getUsageStatistics(hipMemoryPool_t memPool, struct usageStatistics *statistics)
+    {
+        hipMemPoolGetAttribute(memPool, hipMemPoolAttrReservedMemCurrent, &statistics->reservedMemCurrent);
+        hipMemPoolGetAttribute(memPool, hipMemPoolAttrReservedMemHigh, &statistics->reservedMemHigh);
+        hipMemPoolGetAttribute(memPool, hipMemPoolAttrUsedMemCurrent, &statistics->usedMemCurrent);
+        hipMemPoolGetAttribute(memPool, hipMemPoolAttrUsedMemHigh, &statistics->usedMemHigh);
+    }
+
+    // resetting the watermarks will make them take on the current value.
+    void resetStatistics(hipMemoryPool_t memPool)
+    {
+        uint64_t value = 0;
+        hipMemPoolSetAttribute(memPool, hipMemPoolAttrReservedMemHigh, &value);
+        hipMemPoolSetAttribute(memPool, hipMemPoolAttrUsedMemHigh, &value);
+    }
+
+Memory reuse policies
+---------------------
+
+The allocator may reallocate memory as long as it guarantees that compliant memory accesses won't overlap temporally. Turning on and off the following memory pool reuse policy attribute flags can optimize the memory use:
+
```

Comment:
Information changed. "The allocator might reallocate memory as long as the compliant memory accesses are guaranteed not to overlap temporally. To optimize the memory usage, disable or enable the following memory pool reuse policy attribute flags:"

Reply:
Do these flags not control the overlapping of memory accesses? Because in the preceding sentence, we state that memory is allocated as long as memory accesses don't overlap. In that case, we should talk about how to ensure that the accesses are not overlapped. 
Using a sentence like "To optimize the memory usage,...." breaks the link between first and second sentence.

Reply:
It does not control the overlapping, that's always guaranteed.

Reply:
ok